tools/dds/dds-server/dds-server.cpp

```diff
@@ -17,11 +17,39 @@ using namespace tools;
 
 dds_server::dds_server()
     : _running( false )
+    , _trigger_msg_send ( false )
     , _participant( nullptr )
     , _publisher( nullptr )
     , _topic( nullptr )
     , _topic_type( new librealsense::dds::topics::device_info::type )
     , _dds_device_dispatcher( 10 )
+    , _device_info_msg_sender( [this]( dispatcher::cancellable_timer timer ) {
+
+        // We wait until the new reader callback indicate a new reader has joined or until the
+        // active object is stopped
+        std::unique_lock< std::mutex > lock( _device_info_msg_mutex );
+        _device_info_msg_cv.wait( lock, [this]() {
+            return _trigger_msg_send.load() || ! _device_info_msg_sender.is_active();
```

Comment:
We're ignoring `timer` here...

Reply:
Do we need it? I can remove the `timer` name and just leave the type,
I don't see a reason to wake up by timeout

Reply:
It's not by timeout. `timer` is used to wake up when the object dies or is stopped.

Reply:
so you mean replace the condition with `return _trigger_msg_send.load() || timer.was_stopped();`?